### DISCUSSION TITLE
WebRTC player in SwiftUI

### DISCUSSION THREAD

**Discussion by ekiny** [ant-media/Ant-Media-Server#5422] [FAQs]
Hello,

I have seen webrtc publish and play sample but they are all in UIKit enviorement. Is there any sample for SwiftUI?

Thank you.

**Comment by USAMAWIZARD** [comment]
Hi @ekiny , 
Currently its not available , I have created a Feature Request on behalf of you which is available [here](https://github.com/ant-media/Ant-Media-Server/issues/5424) . We will try to add the SwiftUI sample Soon.

**Reply by mekya** [reply]
Hi @ekiny,

Thank you very much for asking this question. 

There is a good demand to the product and it may take some time to create the sample for SwiftUI. 

Would you mind assisting in creating a SwiftUI sample and make a contribution? You may even write a blog post how to create SwiftUI sample with WebRTC iOS SDK and get recognition from the community. We can also publish your post on antmedia.io with your name and recognise your contribution. 

Let me know if it sounds interesting.

Cheers

**Reply by ekiny** [reply]
sure. sounds good.

**Reply by mekya** [reply]
Hi @ekiny,

Let's make sure we're on the same page because you mention that you're waiting for this sample in the issue as mentioned below

Do you proceed to create a SwiftUI sample and make the contribution or you're expecting us to do so? 


https://github.com/ant-media/Ant-Media-Server/issues/5424
> Waiting for this sample.
> It would be great to have video data as buffer image for every frame (pointer, CGImage, CIImage etc...) and seperated audio frame if possible. If SDK does not support frame by frame it is still ok to have the video and audio on avplayer component. I hope we have the sample soon.
> Best regards.

**Reply by ekiny** [reply]
Actually I'm expecting you to create a sample. Thank you.

**Reply by mekya** [reply]
Hi @ekiny,

It will be pleasure on my end. I just need to schedule this work and give the correct priority because we're loaded for the tasks.  
May I learn how many broadcasts and viewers is in your system? You're an enterprise user or after this sample, are you going to become an enterprise user?

**Reply by ekiny** [reply]
I definitely will be a enterprise user. But yes, I need to add this swiftui code my project first.

**Comment by ekiny** [comment]
Hello guys, is there any progress on this task? 
Thank you..

**Reply by USAMAWIZARD** [reply]
Hi ekiny , 
Let me check internally and get back to you. There are some high priority task currently in the pipeline  I am not sure if the this task  is scheduled or not I will provide you an update shortly

**Reply by mekya** [reply]
Let me give a quick update. I think we can ask this to the ecosystem and community to provide a sample for SwiftUI and we can cover the development cost.

**Reply by mekya** [reply]
Let's proceed in this way. Please ask this sample from community or ecosystem and we can cover this cost. @ekiny will you get annual or perpetual license after we provide this sample? Both of them are ok for us.  I mean proceeding in win-win perspective because we don't have resource for this sample and we'll cover the cost separately.   

@USAMAWIZARD  Share the followings on github discussion and ecosystem. you can change title or wording as you wish. 

Freelance: SwiftUI version of WebRTC iOS Sample Applicaiton 

Scope
- Create a project with SwiftUI
- Provide Publish and Play samples step by step in the [same way here](https://antmedia.io/build-and-try-webrtc-ios-app-for-dummies/)
- Provide full working code on github

Send your quote to here or contact@antmedia.io

Cheers

**Comment by USAMAWIZARD** [comment]
@mekya Thanks for the update I have posted An Announcement on Both Eco-System Channel and our [Community Discussion ](https://github.com/orgs/ant-media/discussions/5614)
